diff --git a/ios/Sources/KeyboardPlugin/Keyboard.m b/ios/Sources/KeyboardPlugin/Keyboard.m
index 54a6694fabed1c5ba430ea2514fd26e86a5e933f..53bdfb92f9add156594ee3a8ae671119b3eb90b0 100644
--- a/ios/Sources/KeyboardPlugin/Keyboard.m
+++ b/ios/Sources/KeyboardPlugin/Keyboard.m
@@ -211,7 +211,7 @@ double stageManagerOffset;
 
 - (void)onKeyboardWillHide:(NSNotification *)notification
 {
-  [self setKeyboardHeight:0 delay:0.01];
+  [self setKeyboardHeight:0 delay:0];
   [self resetScrollView];
   hideTimer = [NSTimer scheduledTimerWithTimeInterval:0 repeats:NO block:^(NSTimer * _Nonnull timer) {
     [self.bridge triggerWindowJSEventWithEventName:@"keyboardWillHide"];
@@ -253,8 +253,7 @@ double stageManagerOffset;
     height = 0.0;
   }
 
-  double duration = [[notification.userInfo valueForKey:UIKeyboardAnimationDurationUserInfoKey] doubleValue]+0.2;
-  [self setKeyboardHeight:(int)height delay:duration];
+  [self setKeyboardHeight:(int)height delay:0];
   [self resetScrollView];
 
   NSString * data = [NSString stringWithFormat:@"{ 'keyboardHeight': %d }", (int)height];
@@ -328,7 +327,7 @@ double stageManagerOffset;
         height = screenHeight - paddingBottom;
     }
     
-    [self.bridge evalWithJs: [NSString stringWithFormat:@"(function() { var el = %@; var height = %d; if (el) { el.style.height = height > -1 ? height + 'px' : null; } })()", element, height]];
+    [self.bridge evalWithJs: [NSString stringWithFormat:@"requestAnimationFrame(() => { var el = %@; var height = %d; if (el) { el.style.height = height > -1 ? height + 'px' : null; } })", element, height]];
 }
 
 - (void)_updateFrame
