diff --git a/dist/index.js b/dist/index.js
index efc1633f5c7f9a228863d70c61a0700c5b18c105..0241db7a6f5de1940ebac70dabce5f0226451f24 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -1345,6 +1345,7 @@ const IonTabButton = /*@__PURE__*/ (() => class extends React.Component {
     handleIonTabButtonClick() {
         if (this.props.onClick) {
             this.props.onClick(new CustomEvent('ionTabButtonClick', {
+                cancelable: true,
                 detail: {
                     tab: this.props.tab,
                     href: this.props.href,
@@ -1419,6 +1420,7 @@ class IonTabBarUnwrapped extends React.PureComponent {
         const tabUrl = this.state.tabs[tab];
         if (tabUrl) {
             this.onTabButtonClick(new CustomEvent('ionTabButtonClick', {
+                cancelable: true,
                 detail: {
                     href: tabUrl.currentHref,
                     tab,
@@ -1497,6 +1499,7 @@ class IonTabBarUnwrapped extends React.PureComponent {
              */
             onClickFn(e);
         }
+        if (e.defaultPrevented) return;
         // this.props.onSetCurrentTab(e.detail.tab, this.props.routeInfo);
         // Clicking the current tab will bring you back to the original href
         if (prevActiveTab === e.detail.tab) {
