diff --git a/lib/index.cjs b/lib/index.cjs
index 3eeff6951df74538d0d4e280719b6b49620a4e54..69c73deac360d91d54e791acecdfa4b4ba2ca212 100644
--- a/lib/index.cjs
+++ b/lib/index.cjs
@@ -2,251 +2,251 @@ Object.defineProperty(exports, Symbol.toStringTag, {
     value: "Module"
 });
 
-var e = ([e, t]) => e > t ? [ t, e ] : [ e, t ], t = ([e, t]) => e === t, n = Math.min, {keys: r, is: o} = Object, s = e => "string" == typeof e, c = e => "function" == typeof e, i = c(queueMicrotask) ? queueMicrotask : e => {
-    Promise.resolve().then(e);
-}, a = (e, t = e => x(e) ? e.text : "") => e.children.reduce((e, n, r) => {
-    const o = y(n);
-    return 0 !== r && o && (e += "\n"), e + (o ? n.children.reduce((e, n) => e + t(n), "") : "");
-}, ""), l = (e, t, n) => e.split("\n").map(e => ({
+var t = ([t, e]) => t > e ? [ e, t ] : [ t, e ], e = ([t, e]) => t === e, n = Math.min, {keys: r, is: o} = Object, s = t => "string" == typeof t, c = t => "function" == typeof t, i = c(queueMicrotask) ? queueMicrotask : t => {
+    Promise.resolve().then(t);
+}, a = (t, e = t => y(t) ? t.text : "") => t.children.reduce((t, n, r) => {
+    const o = x(n);
+    return 0 !== r && o && (t += "\n"), t + (o ? n.children.reduce((t, n) => t + e(n), "") : "");
+}, ""), l = (t, e, n) => t.split("\n").map(t => ({
     ...n,
     children: [ {
-        ...t,
-        text: e
+        ...e,
+        text: t
     } ]
-})), f = "delete", u = "insert_text", d = "insert_node", p = "format", h = "set_node_attr", x = e => "text" in e, y = e => "children" in e, _ = (e, t) => {
-    const n = r(e);
-    return n.length === r(t).length && n.every(n => n in t && ("text" === n || o(e[n], t[n])));
-}, k = new WeakMap, g = ({children: e}, t = 0, n = e.length) => {
+})), f = "delete", u = "insert_text", d = "insert_node", p = "format", h = "set_node_attr", y = t => "text" in t, x = t => "children" in t, _ = (t, e) => {
+    const n = r(t);
+    return n.length === r(e).length && n.every(n => n in e && ("text" === n || o(t[n], e[n])));
+}, k = new WeakMap, g = ({children: t}, e = 0, n = t.length) => {
     let r = 0, o = 0;
-    for (let s = t; s < n; s++) {
-        const t = e[s];
-        r += m(t), 0 !== o && y(t) && r++, o++;
+    for (let s = e; s < n; s++) {
+        const e = t[s];
+        r += m(e), 0 !== o && x(e) && r++, o++;
     }
     return r;
-}, m = e => {
-    if (y(e)) {
-        let t = k.get(e);
-        return null == t && k.set(e, t = g(e)), t;
+}, m = t => {
+    if (x(t)) {
+        let e = k.get(t);
+        return null == e && k.set(t, e = g(t)), e;
     }
-    return x(e) ? e.text.length : 1;
-}, b = (e, [t, n]) => {
+    return y(t) ? t.text.length : 1;
+}, b = (t, [e, n]) => {
     let r = 0;
-    for (const o of t) r += g(e, 0, o), 0 !== o && r++, e = e.children[o];
+    for (const o of e) r += g(t, 0, o), 0 !== o && r++, t = t.children[o];
     return r + n;
-}, v = (e, t) => {
-    const n = R(e, t);
+}, v = (t, e) => {
+    const n = R(t, e);
     return [ n.t, n.o ];
-}, E = (e, [t, n]) => [ b(e, t), b(e, n) ], T = (e, [t, n]) => [ v(e, t), v(e, n) ], M = (e, t = 0, n = e.length - 1) => {
-    let r = t + 1;
+}, E = (t, [e, n]) => [ b(t, e), b(t, n) ], T = (t, [e, n]) => [ v(t, e), v(t, n) ], M = (t, e = 0, n = t.length - 1) => {
+    let r = e + 1;
     for (;r <= n; ) {
-        const t = e[r - 1], o = e[r];
-        x(t) && x(o) && _(t, o) ? (e[r - 1] = {
-            ...t,
-            text: t.text + o.text
-        }, e.splice(r, 1), n--) : r++;
+        const e = t[r - 1], o = t[r];
+        y(e) && y(o) && _(e, o) ? (t[r - 1] = {
+            ...e,
+            text: e.text + o.text
+        }, t.splice(r, 1), n--) : r++;
     }
-    for (r = t; r <= n; ) {
-        const t = e[r];
-        x(t) && !t.text && e.length > 1 ? (e.splice(r, 1), n--) : r++;
+    for (r = e; r <= n; ) {
+        const e = t[r];
+        y(e) && !e.text && t.length > 1 ? (t.splice(r, 1), n--) : r++;
     }
-}, S = (e, t = 0, n = e.length - 1) => {
-    let r = t + 1;
+}, S = (t, e = 0, n = t.length - 1) => {
+    let r = e + 1;
     for (;r <= n; ) {
-        const t = e[r - 1], o = e[r];
-        e[r - 1] = A(t, o), e.splice(r, 1), n--;
+        const e = t[r - 1], o = t[r];
+        t[r - 1] = A(e, o), t.splice(r, 1), n--;
     }
-}, w = (e, t, n) => {
-    if (t.length) {
-        const r = e.length;
-        e.push(...t), r && n(e, r - 1, r);
+}, w = (t, e, n) => {
+    if (e.length) {
+        const r = t.length;
+        t.push(...e), r && n(t, r - 1, r);
     }
-}, A = (...e) => ({
-    ...e[0],
-    children: e.reduce((e, t) => (w(e, t.children, M), e), [])
-}), O = ({children: e}, t) => {
-    const n = e.length;
+}, A = (...t) => ({
+    ...t[0],
+    children: t.reduce((t, e) => (w(t, e.children, M), t), [])
+}), O = ({children: t}, e) => {
+    const n = t.length;
     for (let r = 0; r < n; r++) {
-        const n = e[r];
+        const n = t[r];
         let o = m(n);
-        if (y(n) && o++, o > t || o === t && x(n) && !n.text) return {
+        if (x(n) && o++, o > e || o === e && y(n) && !n.text) return {
             i: n,
             l: r,
-            o: t
+            o: e
         };
-        t -= o;
+        e -= o;
     }
     return null;
-}, R = (e, t) => {
+}, R = (t, e) => {
     const n = [];
-    for (;e; ) {
-        const r = O(e, t);
+    for (;t; ) {
+        const r = O(t, e);
         if (!r) break;
         const o = r.i;
-        if (!y(o)) break;
-        t = r.o, e = o, n.push(r.l);
+        if (!x(o)) break;
+        e = r.o, t = o, n.push(r.l);
     }
     return {
-        i: e,
+        i: t,
         t: n,
-        o: t
+        o: e
     };
-}, D = (e, t) => {
-    const n = e.children, r = O(e, t);
+}, D = (t, e) => {
+    const n = t.children, r = O(t, e);
     if (r) {
-        const {i: t, o, l: s} = r;
-        if (y(t)) {
-            const [r, c] = D(t, o), i = n.slice(0, s), a = n.slice(s + 1);
+        const {i: e, o, l: s} = r;
+        if (x(e)) {
+            const [r, c] = D(e, o), i = n.slice(0, s), a = n.slice(s + 1);
             return i.push(r), a.unshift(c), [ {
-                ...e,
+                ...t,
                 children: i
             }, {
-                ...e,
+                ...t,
                 children: a
             } ];
         }
         {
             const r = n.slice(0, s), c = n.slice(s + 1);
-            if (x(t)) {
-                const e = t.text.slice(0, o), n = t.text.slice(o);
-                !e && r.length || r.push({
-                    ...t,
-                    text: e
+            if (y(e)) {
+                const t = e.text.slice(0, o), n = e.text.slice(o);
+                !t && r.length || r.push({
+                    ...e,
+                    text: t
                 }), !n && c.length || c.unshift({
-                    ...t,
+                    ...e,
                     text: n
                 });
-            } else c.unshift(t);
+            } else c.unshift(e);
             return [ {
-                ...e,
+                ...t,
                 children: r
             }, {
-                ...e,
+                ...t,
                 children: c
             } ];
         }
     }
     const o = n[n.length - 1];
-    return [ e, {
-        ...e,
-        children: x(o) ? [ {
+    return [ t, {
+        ...t,
+        children: y(o) ? [ {
             ...o,
             text: ""
         } ] : []
     } ];
-}, B = (e, t, n, r = 0) => {
-    if (r < t.length) {
-        const o = t[r];
-        return ((e, t, n, r) => {
-            const o = e.children.slice();
-            return o.splice(t, n - t + 1, ...r), {
-                ...e,
+}, B = (t, e, n, r = 0) => {
+    if (r < e.length) {
+        const o = e[r];
+        return ((t, e, n, r) => {
+            const o = t.children.slice();
+            return o.splice(e, n - e + 1, ...r), {
+                ...t,
                 children: o
             };
-        })(e, o, o, [ B(e.children[o], t, n, r + 1) ]);
+        })(t, o, o, [ B(t.children[o], e, n, r + 1) ]);
     }
     return n;
-}, I = (e, t, n, r) => {
-    const [o, s] = D(e, t), c = t < n ? D(e, n)[1] : s, i = o.children.slice();
+}, I = (t, e, n, r) => {
+    const [o, s] = D(t, e), c = e < n ? D(t, n)[1] : s, i = o.children.slice();
     return w(i, r, S), w(i, c.children, S), {
-        ...e,
+        ...t,
         children: i
     };
-}, P = (e, t, n) => t >= n ? [] : D(D(e, n)[0], t)[1].children, N = (e, t) => t >= 0 && t <= m(e), L = (e, t) => t.reduce((e, t) => C(e, t), e), C = (e, t) => {
-    switch (t.type) {
+}, P = (t, e, n) => e >= n ? [] : D(D(t, n)[0], e)[1].children, N = (t, e) => e >= 0 && e <= m(t), L = (t, e) => e.reduce((t, e) => C(t, e), t), C = (t, e) => {
+    switch (e.type) {
       case f:
         {
-            const [n, r] = t.range;
-            if (e >= n) return r >= e ? n : e + n - r;
+            const [n, r] = e.range;
+            if (t >= n) return r >= t ? n : t + n - r;
             break;
         }
 
       case u:
         {
-            const {at: n, text: r} = t;
-            if (e >= n) return e + r.length;
+            const {at: n, text: r} = e;
+            if (t >= n) return t + r.length;
             break;
         }
 
       case d:
         {
-            const {at: n, fragment: r} = t;
-            if (e >= n) return e + m({
+            const {at: n, fragment: r} = e;
+            if (t >= n) return t + m({
                 children: r
             });
             break;
         }
     }
-    return e;
-}, H = ([e, t], n) => [ C(e, n), C(t, n) ];
+    return t;
+}, H = ([t, e], n) => [ C(t, n), C(e, n) ];
 
-function U(e, t) {
-    e.apply({
+function U(t, e) {
+    t.apply({
         type: "set_node_attr",
         path: [],
         key: "children",
-        value: t
+        value: e
     });
 }
 
-var F = e => 1 === e.nodeType, V = e => 8 === e.nodeType, W = new Map([ [ "BR", 1 ], [ "WBR", 1 ], [ "AREA", 2 ], [ "BASE", 2 ], [ "DATALIST", 2 ], [ "HEAD", 2 ], [ "LINK", 2 ], [ "META", 2 ], [ "RP", 2 ], [ "SCRIPT", 2 ], [ "STYLE", 2 ], [ "TEMPLATE", 2 ], [ "TITLE", 2 ], [ "COLGROUP", 2 ], [ "HR", 3 ], [ "AUDIO", 3 ], [ "CANVAS", 3 ], [ "EMBED", 3 ], [ "IFRAME", 3 ], [ "IMG", 3 ], [ "OBJECT", 3 ], [ "PICTURE", 3 ], [ "VIDEO", 3 ], [ "SVG", 3 ], [ "MATH", 3 ], [ "BUTTON", 3 ], [ "INPUT", 3 ], [ "METER", 3 ], [ "PROGRESS", 3 ], [ "SELECT", 3 ], [ "TEXTAREA", 3 ] ]), z = new Set([ "DIV", "H1", "H2", "H3", "H4", "H5", "H6", "P", "PRE", "LI", "DT", "DD", "TH", "TD" ]), G = e => z.has(e.tagName), q = (e, t) => e.compareDocumentPosition(t), J = e => e.ownerDocument, j = e => J(e).getSelection(), X = (e, t) => {
-    if (e.rangeCount) {
-        const n = e.getRangeAt(0);
-        if (t.contains(n.commonAncestorContainer)) return n;
+var F = t => 1 === t.nodeType, V = t => 8 === t.nodeType, W = new Map([ [ "BR", 1 ], [ "WBR", 1 ], [ "AREA", 2 ], [ "BASE", 2 ], [ "DATALIST", 2 ], [ "HEAD", 2 ], [ "LINK", 2 ], [ "META", 2 ], [ "RP", 2 ], [ "SCRIPT", 2 ], [ "STYLE", 2 ], [ "TEMPLATE", 2 ], [ "TITLE", 2 ], [ "COLGROUP", 2 ], [ "HR", 3 ], [ "AUDIO", 3 ], [ "CANVAS", 3 ], [ "EMBED", 3 ], [ "IFRAME", 3 ], [ "IMG", 3 ], [ "OBJECT", 3 ], [ "PICTURE", 3 ], [ "VIDEO", 3 ], [ "SVG", 3 ], [ "MATH", 3 ], [ "BUTTON", 3 ], [ "INPUT", 3 ], [ "METER", 3 ], [ "PROGRESS", 3 ], [ "SELECT", 3 ], [ "TEXTAREA", 3 ] ]), z = new Set([ "DIV", "H1", "H2", "H3", "H4", "H5", "H6", "P", "PRE", "LI", "DT", "DD", "TH", "TD" ]), G = t => z.has(t.tagName), q = (t, e) => t.compareDocumentPosition(e), J = t => t.ownerDocument, j = t => J(t).getSelection(), X = (t, e) => {
+    if (t.rangeCount) {
+        const n = t.getRangeAt(0);
+        if (e.contains(n.commonAncestorContainer)) return n;
     }
-}, Y = (e, t, n, [r, o], s, c) => {
-    const i = j(t);
-    if (c || X(i, t)) {
-        const c = 0 === s, a = s > 0, l = a ? r : o, f = Z(t, n, a ? o : r), u = c ? f : Z(t, n, l), d = e.createRange(), [p, h] = f, [x, y] = u;
-        F(p) && t !== p ? h < 1 ? d.setStartBefore(p) : d.setStartAfter(p) : d.setStart(p, h), 
-        F(x) && t !== x ? y < 1 ? d.setEndBefore(x) : d.setEndAfter(x) : d.setEnd(x, y), 
+}, Y = (t, e, n, [r, o], s, c) => {
+    const i = j(e);
+    if (c || X(i, e)) {
+        const c = 0 === s, a = s > 0, l = a ? r : o, f = Z(e, n, a ? o : r), u = c ? f : Z(e, n, l), d = t.createRange(), [p, h] = f, [y, x] = u;
+        F(p) && e !== p ? h < 1 ? d.setStartBefore(p) : d.setStartAfter(p) : d.setStart(p, h), 
+        F(y) && e !== y ? x < 1 ? d.setEndBefore(y) : d.setEndAfter(y) : d.setEnd(y, x), 
         i.removeAllRanges(), i.addRange(d), a && (i.collapseToEnd(), i.extend(d.startContainer, d.startOffset));
     }
-}, Z = (e, t, [n, r]) => t(({u: t, p: o, h: s, _: c}) => {
+}, Z = (t, e, [n, r]) => e(({u: e, p: o, h: s, _: c}) => {
     let i, a = 0;
-    for (;i = t(); ) if (4 === i) {
-        if (a < n.length) for (let e = n[a++]; e > 0; e--) o();
+    for (;i = e(); ) if (4 === i) {
+        if (a < n.length) for (let t = n[a++]; t > 0; t--) o();
     } else {
-        const e = c();
-        if (r <= e) return [ s(), r ];
-        r -= e;
+        const t = c();
+        if (r <= t) return [ s(), r ];
+        r -= t;
     }
-    return [ e, 0 ];
-}, e), K = (e, t, r, o) => {
+    return [ t, 0 ];
+}, t), K = (t, e, r, o) => {
     let s = !0;
-    if (e === r && !r.hasChildNodes()) return [ [ 0 ], 0 ];
+    if (t === r && !r.hasChildNodes()) return [ [ 0 ], 0 ];
     if (F(r) && r.hasChildNodes()) {
-        const e = n(o, r.childNodes.length - 1);
-        r = r.childNodes[e], s = e === o, o = 0;
+        const t = n(o, r.childNodes.length - 1);
+        r = r.childNodes[t], s = t === o, o = 0;
     }
-    return t(({u: n, k: c, m: i, v: a, h: l, _: f, T: u}) => {
+    return e(({u: n, k: c, m: i, v: a, h: l, _: f, T: u}) => {
         c(r), 4 !== u() && i();
-        const d = t(() => {
-            const t = [];
-            for (;u() && l() !== e; ) {
-                let e = 0;
-                for (;a(), u(); ) e++;
-                t.unshift(e), i();
+        const d = e(() => {
+            const e = [];
+            for (;u() && l() !== t; ) {
+                let t = 0;
+                for (;a(), u(); ) t++;
+                e.unshift(t), i();
             }
-            return t.length ? t : [ 0 ];
+            return e.length ? e : [ 0 ];
         });
         let p = 0;
         for (;n(); ) {
-            const e = q(r, l());
-            if (0 === e || 16 & e) {
+            const t = q(r, l());
+            if (0 === t || 16 & t) {
                 if (s) break;
-            } else if (4 & e) break;
+            } else if (4 & t) break;
             p += f();
         }
         return [ d, p + o ];
-    }, e);
-}, Q = (e, t, {startOffset: n, startContainer: r, endOffset: o, endContainer: s}) => {
-    const c = K(e, t, r, n);
-    return [ c, r === s && n === o ? c : K(e, t, s, o) ];
-}, $ = "undefined" != typeof navigator && /Mac|iP(hone|od|ad)/.test(navigator.platform), ee = (e, t) => {
-    const n = e.endsWith("+"), r = (n ? e.slice(0, -2) : e).split("+"), o = n ? "+" : r.pop(), s = "Space" === o ? " " : o.toLowerCase();
+    }, t);
+}, Q = (t, e, {startOffset: n, startContainer: r, endOffset: o, endContainer: s}) => {
+    const c = K(t, e, r, n);
+    return [ c, r === s && n === o ? c : K(t, e, s, o) ];
+}, $ = "undefined" != typeof navigator && /Mac|iP(hone|od|ad)/.test(navigator.platform), tt = (t, e) => {
+    const n = t.endsWith("+"), r = (n ? t.slice(0, -2) : t).split("+"), o = n ? "+" : r.pop(), s = "Space" === o ? " " : o.toLowerCase();
     let c = !1, i = !1, a = !1, l = !1;
-    return r.forEach(e => {
-        switch (e) {
+    return r.forEach(t => {
+        switch (t) {
           case "Ctrl":
             i = !0;
             break;
@@ -266,45 +266,45 @@ var F = e => 1 === e.nodeType, V = e => 8 === e.nodeType, W = new Map([ [ "BR",
           case "Shift":
             c = !0;
         }
-    }), e => {
-        if (e.key.toLowerCase() === s && i === e.ctrlKey && a === e.metaKey && c === e.shiftKey && l === e.altKey) return t(e), 
+    }), t => {
+        if (t.key.toLowerCase() === s && i === t.ctrlKey && a === t.metaKey && c === t.shiftKey && l === t.altKey) return e(t), 
         !0;
     };
 };
 
-function te(e) {
-    let t = 0, n = 0, r = !1;
-    const s = Date.now, c = [ [ e.doc, e.selection, [] ] ], i = () => c[t], a = () => t > 0, l = () => t < c.length - 1, f = () => {
+function et(t) {
+    let e = 0, n = 0, r = !1;
+    const s = Date.now, c = [ [ t.doc, t.selection, [] ] ], i = () => c[e], a = () => e > 0, l = () => e < c.length - 1, f = () => {
         if (a()) {
             const n = i()[1];
-            t--;
-            const s = e.doc;
-            r = !0, e.exec(U, i()[0].children), r = !1, o(s, e.doc) || (e.selection = n);
+            e--;
+            const s = t.doc;
+            r = !0, t.exec(U, i()[0].children), r = !1, o(s, t.doc) || (t.selection = n);
         }
     }, u = () => {
         if (l()) {
-            t++;
-            const [n, s, c] = i(), a = e.doc;
-            r = !0, e.exec(U, n.children), r = !1, o(a, e.doc) || (e.selection = [ L(s[0], c), L(s[1], c) ]);
+            e++;
+            const [n, s, c] = i(), a = t.doc;
+            r = !0, t.exec(U, n.children), r = !1, o(a, t.doc) || (t.selection = [ L(s[0], c), L(s[1], c) ]);
         }
     };
-    e.hook("apply", (i, a) => {
+    t.hook("apply", (i, a) => {
         if (r) return;
-        const l = e.doc, f = e.selection;
+        const l = t.doc, f = t.selection;
         a(i);
-        const u = e.doc;
+        const u = t.doc;
         if (!o(l, u)) {
-            const e = s();
-            if (0 === t || e - n >= 500) {
-                t++;
-                const e = [ l, f, [] ];
-                t >= c.length ? c.push(e) : c[t] = e;
+            const t = s();
+            if (0 === e || t - n >= 500) {
+                e++;
+                const t = [ l, f, [] ];
+                e >= c.length ? c.push(t) : c[e] = t;
             }
-            n = e, c[t][0] = u, c[t][2].push(i), c.length > t + 1 && (c.length = t + 1), t > 500 && (t--, 
+            n = t, c[e][0] = u, c[e][2].push(i), c.length > e + 1 && (c.length = e + 1), e > 500 && (e--, 
             c.shift());
         }
-    }), e.hook("keyboard", ee("Mod+Z", f)), e.hook("keyboard", ee("Shift+Mod+Z", u)), 
-    e.set(te, {
+    }), t.hook("keyboard", tt("Mod+Z", f)), t.hook("keyboard", tt("Shift+Mod+Z", u)), 
+    t.set(et, {
         undo: f,
         redo: u,
         undoable: a,
@@ -312,63 +312,63 @@ function te(e) {
     });
 }
 
-var ne = [], re = () => {}, oe = e => {
-    throw Error(e);
-}, se = ({doc: n, readonly: r = !1, schema: a, isBlock: y = G, onWarn: _ = console.warn, onError: k = oe}) => {
+var nt = [], rt = () => {}, ot = t => {
+    throw Error(t);
+}, st = ({doc: n, readonly: r = !1, schema: a, isBlock: x = G, onWarn: _ = console.warn, onError: k = ot}) => {
     let g = [ 0, 0 ];
-    const m = e => {
+    const m = t => {
         if (!a) return _("An unsafe operation was detected. We recommend using schema option."), 
         !0;
-        const t = a["~standard"].validate(e);
-        if (t instanceof Promise) k("async validate is not supported"); else {
-            if (!t.issues) return !0;
-            _(t.issues.map(e => e.message).join("\n"));
+        const e = a["~standard"].validate(t);
+        if (e instanceof Promise) k("async validate is not supported"); else {
+            if (!e.issues) return !0;
+            _(e.issues.map(t => t.message).join("\n"));
         }
         return !1;
     };
     a && !m(n) && k("Invalid document");
-    const v = new Map, M = new WeakMap, S = e => v.get(e) || ne, w = new Map, A = new Set, D = e => {
-        const t = w.get(e);
-        t && (A.size || i(() => {
-            A.forEach(e => {
-                e();
+    const v = new Map, M = new WeakMap, S = t => v.get(t) || nt, w = new Map, A = new Set, D = t => {
+        const e = w.get(t);
+        e && (A.size || i(() => {
+            A.forEach(t => {
+                t();
             }), A.clear();
-        }), t.forEach(e => {
-            A.add(e);
+        }), e.forEach(t => {
+            A.add(t);
         }));
-    }, C = e => {
-        if (!r) if (Array.isArray(e)) for (const t of e) U(t); else U(e);
-        return z;
-    }, U = e => {
-        const t = n, r = S("apply"), s = r.length;
+    }, C = t => {
+        if (!r) if (Array.isArray(t)) for (const e of t) U(e); else U(t);
+        return Z;
+    }, U = t => {
+        const e = n, r = S("apply"), s = r.length;
         let c = 0;
         const i = () => {
             if (c < s) {
-                const t = c;
-                r[c](e, a), t === c && a();
+                const e = c;
+                r[c](t, a), e === c && a();
             } else if (c === s) {
                 c++;
                 try {
-                    const [t, r] = ((e, t, n) => {
+                    const [e, r] = ((t, e, n) => {
                         switch (n.type) {
                           case f:
                             {
                                 const [r, o] = n.range;
-                                N(e, r) && N(e, o) && r < o && (e = I(e, r, o, []), t = H(t, n));
+                                N(t, r) && N(t, o) && r < o && (t = I(t, r, o, []), e = H(e, n));
                                 break;
                             }
 
                           case u:
                             {
                                 const {at: r, text: o} = n;
-                                if (N(e, r) && o) {
-                                    const {i: s, o: c} = R(e, r), i = O(s, c - 1);
+                                if (N(t, r) && o) {
+                                    const {i: s, o: c} = R(t, r), i = O(s, c - 1);
                                     let a;
                                     if (i) {
-                                        const e = i.i;
-                                        x(e) && (a = e);
+                                        const t = i.i;
+                                        y(t) && (a = t);
                                     }
-                                    e = I(e, r, r, l(o, a, s)), t = H(t, n);
+                                    t = I(t, r, r, l(o, a, s)), e = H(e, n);
                                 }
                                 break;
                             }
@@ -376,149 +376,151 @@ var ne = [], re = () => {}, oe = e => {
                           case d:
                             {
                                 const {at: r, fragment: o} = n;
-                                N(e, r) && o.length && (e = I(e, r, r, o), t = H(t, n));
+                                N(t, r) && o.length && (t = I(t, r, r, o), e = H(e, n));
                                 break;
                             }
 
                           case p:
                             {
-                                const {range: [t, r], key: o, value: s} = n;
-                                if (N(e, t) && N(e, r) && t <= r) if (t === r) {
-                                    const {i: {children: n}, t: r} = R(e, t);
+                                const {range: [e, r], key: o, value: s} = n;
+                                if (N(t, e) && N(t, r) && e <= r) if (e === r) {
+                                    const {i: {children: n}, t: r} = R(t, e);
                                     if (1 === n.length) {
-                                        const t = n[0];
-                                        x(t) && !t.text && (e = B(e, [ ...r, 0 ], {
-                                            ...t,
+                                        const e = n[0];
+                                        y(e) && !e.text && (t = B(t, [ ...r, 0 ], {
+                                            ...e,
                                             [o]: s
                                         }));
                                     }
-                                } else e = I(e, t, r, P(e, t, r).map(e => ({
-                                    ...e,
-                                    children: e.children.map(e => x(e) ? {
-                                        ...e,
+                                } else t = I(t, e, r, P(t, e, r).map(t => ({
+                                    ...t,
+                                    children: t.children.map(t => y(t) ? {
+                                        ...t,
                                         [o]: s
-                                    } : e)
+                                    } : t)
                                 })));
                                 break;
                             }
 
                           case h:
                             {
-                                const {path: t, key: r, value: o} = n, s = ((e, t) => {
-                                    for (let n = 0; n < t.length; n++) e = e.children[t[n]];
-                                    return e;
-                                })(e, t);
-                                s && (e = B(e, t, {
+                                const {path: e, key: r, value: o} = n, s = ((t, e) => {
+                                    for (let n = 0; n < e.length; n++) t = t.children[e[n]];
+                                    return t;
+                                })(t, e);
+                                s && (t = B(t, e, {
                                     ...s,
                                     [r]: o
                                 }));
                                 break;
                             }
                         }
-                        return [ e, t ];
-                    })(n, g, e);
-                    (({type: e}) => e === d || e === p || e === h)(e) && !m(t) || (n = t, V(r));
-                } catch (t) {
-                    _("rollback operation: " + t);
+                        return [ t, e ];
+                    })(n, g, t);
+                    (({type: t}) => t === d || t === p || t === h)(t) && !m(e) || (n = e, z(r));
+                } catch (e) {
+                    _("rollback operation: " + e);
                 }
             }
-        }, a = t => {
-            t && (e = t), c++, i();
+        }, a = e => {
+            e && (t = e), c++, i();
         };
-        i(), o(t, n) || D("change");
-    }, V = e => {
-        !((e, [t, n]) => N(e, t) && N(e, n))(n, e) || g[0] === e[0] && g[1] === e[1] || (g = e, 
+        i(), o(e, n) || D("change");
+    };
+    let V;
+    const z = t => {
+        !((t, [e, n]) => N(t, e) && N(t, n))(n, t) || g[0] === t[0] && g[1] === t[1] || (g = t, 
         D("selectionchange"));
-    }, z = {
+    }, Z = {
         get doc() {
             return n;
         },
         get selection() {
             return g;
         },
-        set selection(e) {
-            V(e);
+        set selection(t) {
+            z(t), V?.();
         },
         get readonly() {
             return r;
         },
-        set readonly(e) {
-            r = e, D("readonly");
+        set readonly(t) {
+            r = t, D("readonly");
         },
         apply: C,
-        on: (e, t) => {
-            let n = w.get(e);
-            return n || w.set(e, n = new Set), n.add(t), () => {
-                n.delete(t);
+        on: (t, e) => {
+            let n = w.get(t);
+            return n || w.set(t, n = new Set), n.add(e), () => {
+                n.delete(e);
             };
         },
-        hook: (e, t) => {
-            let n = v.get(e);
-            return n || v.set(e, n = []), n.push(t), () => {
-                const e = n.indexOf(t);
-                -1 !== e && n.splice(e, 1);
+        hook: (t, e) => {
+            let n = v.get(t);
+            return n || v.set(t, n = []), n.push(e), () => {
+                const t = n.indexOf(e);
+                -1 !== t && n.splice(t, 1);
             };
         },
-        exec: (e, ...t) => {
-            const n = e(z, ...t);
-            return void 0 === n ? z : n;
+        exec: (t, ...e) => {
+            const n = t(Z, ...e);
+            return void 0 === n ? Z : n;
         },
-        get: e => (M.has(e) || k("No value found for key"), M.get(e)),
-        set: (e, t) => (M.set(e, t), z),
+        get: t => (M.has(t) || k("No value found for key"), M.get(t)),
+        set: (t, e) => (M.set(t, e), Z),
         input: o => {
             if (!window.InputEvent || !c(InputEvent.prototype.getTargetRanges)) return _("beforeinput event is not supported."), 
-            re;
+            rt;
             const {contentEditable: i, role: a, ariaMultiLine: l, ariaReadOnly: f} = o, u = o.style.whiteSpace;
             o.role = "textbox", o.style.whiteSpace = "pre-wrap", o.ariaMultiLine = "true";
-            let d = !1, p = null, h = !1, x = !1, k = !1;
-            const m = J(o), v = ((e, t) => {
+            let d = !1, p = null, h = !1, y = !1, k = !1, m = !1;
+            const v = J(o), M = ((t, e) => {
                 let n = null, r = null, o = null;
                 const s = () => {
                     if (null != o) return o;
-                    if (r) if ((e => 3 === e.nodeType)(r)) {
-                        const e = r.data;
-                        if (e) return o = "\n" === e ? i() ? 3 : 5 : 1;
+                    if (r) if ((t => 3 === t.nodeType)(r)) {
+                        const t = r.data;
+                        if (t) return o = "\n" === t ? i() ? 3 : 5 : 1;
                     } else if (F(r)) {
                         if ("false" === r.contentEditable) return o = 2;
                         {
-                            const e = W.get(r.tagName);
-                            if (null != e) return o = 1 === e ? i() ? 3 : 5 : 3 === e ? 2 : 6;
-                            if (t(r)) return o = 4;
+                            const t = W.get(r.tagName);
+                            if (null != t) return o = 1 === t ? i() ? 3 : 5 : 3 === t ? 2 : 6;
+                            if (e(r)) return o = 4;
                         }
-                    } else if (!(e => 11 === e.nodeType)(r)) return o = 6;
+                    } else if (!(t => 11 === t.nodeType)(r)) return o = 6;
                     return o = 0;
                 }, c = () => {
-                    const e = s();
-                    if (o = null, 2 === e || 6 === e) {
-                        const e = r;
-                        if (r = n.nextSibling(), !r) for (;(r = n.nextNode()) && e.contains(r); ) ;
+                    const t = s();
+                    if (o = null, 2 === t || 6 === t) {
+                        const t = r;
+                        if (r = n.nextSibling(), !r) for (;(r = n.nextNode()) && t.contains(r); ) ;
                         return r;
                     }
                     return r = n.nextNode();
                 }, i = () => {
-                    const e = r.parentNode;
+                    const t = r.parentNode;
                     return l(() => {
                         for (o = 0; c(); ) {
                             if (s()) return !0;
-                            if (!e.contains(r)) break;
+                            if (!t.contains(r)) break;
                         }
                         return !1;
                     });
                 }, a = {
                     u: () => {
                         for (;c(); ) {
-                            const e = s();
-                            if (e && 6 !== e) return e;
+                            const t = s();
+                            if (t && 6 !== t) return t;
                         }
                     },
                     T: s,
                     h: () => r,
                     _: () => {
-                        const e = s();
-                        return 1 === e ? r.data.length : 2 === e ? 1 : 0;
+                        const t = s();
+                        return 1 === t ? r.data.length : 2 === t ? 1 : 0;
                     },
-                    k: e => {
-                        o = null, n.currentNode = r = e;
+                    k: t => {
+                        o = null, n.currentNode = r = t;
                     },
                     v: () => {
                         for (;(o = null) || (r = n.previousSibling()); ) if (4 === s()) return;
@@ -529,61 +531,61 @@ var ne = [], re = () => {}, oe = e => {
                     m: () => {
                         for (;(o = null) || (r = n.parentNode()); ) if (4 === s()) return;
                     }
-                }, l = (t, s) => {
+                }, l = (e, s) => {
                     const c = n, i = r, l = o;
                     try {
-                        return n || (n = e.createTreeWalker(r = s, 5)), t(a);
+                        return n || (n = t.createTreeWalker(r = s, 5)), e(a);
                     } finally {
                         n = c, r = i, o = l, n && i && (n.currentNode = i);
                     }
                 };
                 return l;
-            })(m, y), M = () => {
+            })(v, x), w = () => {
                 o.contentEditable = r ? "false" : "true", o.ariaReadOnly = r ? "true" : null;
             };
-            M();
-            const w = z.on("change", () => {
-                x || requestAnimationFrame(() => {
-                    x || o.focus({
+            w();
+            const A = Z.on("change", () => {
+                y || requestAnimationFrame(() => {
+                    y || o.focus({
                         preventScroll: !0
                     });
                 });
-            }), A = z.on("readonly", M), O = e => {
-                for (const t of S("paste")) {
-                    const n = t(e);
+            }), O = Z.on("readonly", w), R = t => {
+                for (const e of S("paste")) {
+                    const n = e(t);
                     if (n) return n;
                 }
                 _("failed to serialize pasted data");
-            }, R = ((e, t) => {
+            }, D = ((t, e) => {
                 let n = !1, r = [];
-                const o = e => {
-                    n && r.push(...e);
-                }, s = new MutationObserver(e => {
-                    o(e), n || t();
+                const o = t => {
+                    n && r.push(...t);
+                }, s = new MutationObserver(t => {
+                    o(t), n || e();
                 }), c = () => {
                     o(s.takeRecords());
                 }, i = () => {
                     c();
-                    const e = r;
-                    return r = [], e;
+                    const t = r;
+                    return r = [], t;
                 };
-                return s.observe(e, {
+                return s.observe(t, {
                     characterData: !0,
                     characterDataOldValue: !0,
                     childList: !0,
                     subtree: !0
                 }), {
-                    M: e => {
-                        let t;
-                        for (;t = e.pop(); ) if ("childList" === t.type) {
-                            const {target: e, removedNodes: n, addedNodes: r, nextSibling: o} = t;
-                            for (let t = n.length - 1; t >= 0; t--) e.insertBefore(n[t], o);
-                            for (let t = r.length - 1; t >= 0; t--) e.removeChild(r[t]);
-                        } else t.target.data = t.oldValue;
+                    M: t => {
+                        let e;
+                        for (;e = t.pop(); ) if ("childList" === e.type) {
+                            const {target: t, removedNodes: n, addedNodes: r, nextSibling: o} = e;
+                            for (let e = n.length - 1; e >= 0; e--) t.insertBefore(n[e], o);
+                            for (let e = r.length - 1; e >= 0; e--) t.removeChild(r[e]);
+                        } else e.target.data = e.oldValue;
                         i();
                     },
-                    S: e => {
-                        !n && e && c(), n = e;
+                    S: t => {
+                        !n && t && c(), n = t;
                     },
                     A: i,
                     O: () => {
@@ -591,37 +593,45 @@ var ne = [], re = () => {}, oe = e => {
                     }
                 };
             })(o, () => {
-                Y(m, o, v, T(n, g), g[0] - g[1]);
-            }), D = () => {
-                V(E(n, ((e, t) => {
-                    const n = j(e), r = X(n, e);
+                Y(v, o, M, T(n, g), g[0] - g[1]);
+            }), B = () => {
+                z(E(n, ((t, e) => {
+                    const n = j(t), r = X(n, t);
                     if (!r) return [ [ [ 0 ], 0 ], [ [ 0 ], 0 ] ];
-                    const o = Q(e, t, r), s = q(n.anchorNode, n.focusNode);
+                    const o = Q(t, e, r), s = q(n.anchorNode, n.focusNode);
                     return (0 === s ? n.anchorOffset > n.focusOffset : 2 & s) ? [ o[1], o[0] ] : o;
-                })(o, v)));
-            }, B = () => {
-                const e = R.A();
-                R.S(!1), e.length && (R.M(e), m.removeEventListener("selectionchange", $), Y(m, o, v, T(n, g), g[0] - g[1], !0), 
-                m.addEventListener("selectionchange", $)), p && (V(p[1]), C(p[0]), p = null), h = !1;
-            }, I = e => {
-                if (!h) for (const t of S("keyboard")) if (t(e)) return e.preventDefault(), void R.S(!1);
-            }, P = () => {
-                h || B();
-            }, N = e => {
-                e.preventDefault();
-                const r = e.inputType;
+                })(o, M)));
+            };
+            V = () => {
+                !y || m || h || Y(v, o, M, T(n, g), g[0] - g[1]);
+            };
+            const I = () => {
+                m = !0;
+                const t = D.A();
+                D.S(!1), t.length && (D.M(t), v.removeEventListener("selectionchange", nt), Y(v, o, M, T(n, g), g[0] - g[1], !0), 
+                v.addEventListener("selectionchange", nt)), p && (z(p[1]), C(p[0]), p = null), h = !1, 
+                m = !1;
+            }, P = Z.hook("apply", (t, e) => {
+                h && !m && I(), e();
+            }), N = t => {
+                if (!h) for (const e of S("keyboard")) if (e(t)) return t.preventDefault(), void D.S(!1);
+            }, H = () => {
+                h || I();
+            }, U = t => {
+                t.preventDefault();
+                const r = t.inputType;
                 if (r.startsWith("format")) return;
                 if ("historyUndo" === r || "historyRedo" === r) return;
-                h ? R.S(!0) : D();
-                const s = e.getTargetRanges()[0];
+                h ? D.S(!0) : B();
+                const s = t.getTargetRanges()[0];
                 if (s) {
-                    const c = E(n, Q(o, v, s));
-                    let i, a = "insertParagraph" === r || "insertLineBreak" === r ? "\n" : e.data;
+                    const c = E(n, Q(o, M, s));
+                    let i, a = "insertParagraph" === r || "insertLineBreak" === r ? "\n" : t.data;
                     if (null == a) {
-                        const t = e.dataTransfer;
-                        t && (a = t.getData("text/plain"));
+                        const e = t.dataTransfer;
+                        e && (a = e.getData("text/plain"));
                     }
-                    p || (p = [ [], g ]), i = p[0], t(c) || i.push({
+                    p || (p = [ [], g ]), i = p[0], e(c) || i.push({
                         type: "delete",
                         range: c
                     }), a && i.push({
@@ -630,33 +640,33 @@ var ne = [], re = () => {}, oe = e => {
                         text: a
                     });
                 }
-                h || B();
-            }, H = () => {
-                h || D(), h = !0;
-            }, U = () => {
-                B();
+                h || I();
             }, G = () => {
-                x = !0, D();
-            }, Z = () => {
-                x = !1;
+                h || B(), h = !0;
             }, $ = () => {
-                !x || h || k || D();
-            }, ee = e => {
-                if (D(), !t(g)) for (const t of S("copy")) t(e);
-            }, te = e => {
-                e.preventDefault(), ee(e.clipboardData);
-            }, ne = t => {
-                t.preventDefault(), r || (ee(t.clipboardData), C({
+                I();
+            }, tt = () => {
+                y = !0, B();
+            }, et = () => {
+                y = !1;
+            }, nt = () => {
+                !y || h || k || B();
+            }, ot = t => {
+                if (B(), !e(g)) for (const e of S("copy")) e(t);
+            }, st = t => {
+                t.preventDefault(), ot(t.clipboardData);
+            }, ct = e => {
+                e.preventDefault(), r || (ot(e.clipboardData), C({
                     type: "delete",
-                    range: e(g)
+                    range: t(g)
                 }));
-            }, oe = t => {
-                t.preventDefault();
-                const n = O(t.clipboardData);
+            }, it = e => {
+                e.preventDefault();
+                const n = R(e.clipboardData);
                 if (n) {
-                    const t = e(g), r = t[0], o = [ {
+                    const e = t(g), r = e[0], o = [ {
                         type: "delete",
-                        range: t
+                        range: e
                     } ];
                     o.push(s(n) ? {
                         type: "insert_text",
@@ -668,28 +678,28 @@ var ne = [], re = () => {}, oe = e => {
                         fragment: n
                     }), C(o);
                 }
-            }, se = t => {
-                t.preventDefault();
-                const r = t.dataTransfer, c = ((e, t, {clientX: n, clientY: r}) => {
-                    const o = J(e);
+            }, at = e => {
+                e.preventDefault();
+                const r = e.dataTransfer, c = ((t, e, {clientX: n, clientY: r}) => {
+                    const o = J(t);
                     if (o.caretPositionFromPoint) {
                         const s = o.caretPositionFromPoint(n, r);
-                        if (s) return K(e, t, s.offsetNode, s.offset);
+                        if (s) return K(t, e, s.offsetNode, s.offset);
                     } else if (o.caretRangeFromPoint) {
                         const s = o.caretRangeFromPoint(n, r);
-                        if (s) return K(e, t, s.startContainer, s.startOffset);
+                        if (s) return K(t, e, s.startContainer, s.startOffset);
                     }
-                })(o, v, t);
+                })(o, M, e);
                 if (r && c) {
-                    let t;
+                    let e;
                     const o = [];
                     k && o.push({
                         type: "delete",
-                        range: e(g)
+                        range: t(g)
                     });
-                    const i = O(r);
+                    const i = R(r);
                     if (i) {
-                        const e = b(n, c), r = L(e, o);
+                        const t = b(n, c), r = L(t, o);
                         o.push(s(i) ? {
                             type: "insert_text",
                             at: r,
@@ -698,98 +708,98 @@ var ne = [], re = () => {}, oe = e => {
                             type: "insert_node",
                             at: r,
                             fragment: i
-                        }), t = [ r, L(e, o) ];
+                        }), e = [ r, L(t, o) ];
                     }
-                    C(o), t && V(t);
+                    C(o), e && z(e);
                 }
-            }, ce = e => {
-                k = !0, ee(e.dataTransfer);
-            }, ie = () => {
+            }, lt = t => {
+                k = !0, ot(t.dataTransfer);
+            }, ft = () => {
                 k = !1;
             };
-            m.addEventListener("selectionchange", $), o.addEventListener("keydown", I), o.addEventListener("input", P), 
-            o.addEventListener("beforeinput", N), o.addEventListener("compositionstart", H), 
-            o.addEventListener("compositionend", U), o.addEventListener("focus", G), o.addEventListener("blur", Z), 
-            o.addEventListener("copy", te), o.addEventListener("cut", ne), o.addEventListener("paste", oe), 
-            o.addEventListener("drop", se), o.addEventListener("dragstart", ce), o.addEventListener("dragend", ie);
-            const ae = [];
-            return S("mount").forEach(e => {
-                const t = e(o, v);
-                t && ae.push(t);
+            v.addEventListener("selectionchange", nt), o.addEventListener("keydown", N), o.addEventListener("input", H), 
+            o.addEventListener("beforeinput", U), o.addEventListener("compositionstart", G), 
+            o.addEventListener("compositionend", $), o.addEventListener("focus", tt), o.addEventListener("blur", et), 
+            o.addEventListener("copy", st), o.addEventListener("cut", ct), o.addEventListener("paste", it), 
+            o.addEventListener("drop", at), o.addEventListener("dragstart", lt), o.addEventListener("dragend", ft);
+            const ut = [];
+            return S("mount").forEach(t => {
+                const e = t(o, M);
+                e && ut.push(e);
             }), () => {
-                d || (d = !0, w(), A(), o.contentEditable = i, o.role = a, o.ariaMultiLine = l, 
-                o.ariaReadOnly = f, o.style.whiteSpace = u, R.O(), m.removeEventListener("selectionchange", $), 
-                o.removeEventListener("keydown", I), o.removeEventListener("input", P), o.removeEventListener("beforeinput", N), 
-                o.removeEventListener("compositionstart", H), o.removeEventListener("compositionend", U), 
-                o.removeEventListener("focus", G), o.removeEventListener("blur", Z), o.removeEventListener("copy", te), 
-                o.removeEventListener("cut", ne), o.removeEventListener("paste", oe), o.removeEventListener("drop", se), 
-                o.removeEventListener("dragstart", ce), o.removeEventListener("dragend", ie), ae.forEach(e => {
-                    e();
+                d || (d = !0, A(), O(), o.contentEditable = i, o.role = a, o.ariaMultiLine = l, 
+                o.ariaReadOnly = f, o.style.whiteSpace = u, D.O(), P(), V = void 0, v.removeEventListener("selectionchange", nt), 
+                o.removeEventListener("keydown", N), o.removeEventListener("input", H), o.removeEventListener("beforeinput", U), 
+                o.removeEventListener("compositionstart", G), o.removeEventListener("compositionend", $), 
+                o.removeEventListener("focus", tt), o.removeEventListener("blur", et), o.removeEventListener("copy", st), 
+                o.removeEventListener("cut", ct), o.removeEventListener("paste", it), o.removeEventListener("drop", at), 
+                o.removeEventListener("dragstart", lt), o.removeEventListener("dragend", ft), ut.forEach(t => {
+                    t();
                 }));
             };
         }
     };
-    return z.exec(te), z;
+    return Z.exec(et), Z;
 };
 
-function ce(t, n) {
+function ct(e, n) {
     const r = n && n.serializer;
-    t.hook("copy", n => {
+    e.hook("copy", n => {
         n.setData("text/plain", a({
-            children: P(t.doc, ...e(t.selection))
+            children: P(e.doc, ...t(e.selection))
         }, r));
-    }), t.hook("paste", e => e.getData("text/plain"));
+    }), e.hook("paste", t => t.getData("text/plain"));
 }
 
-var ie = "application/x-editate-editor";
+var it = "application/x-editate-editor";
 
-function ae(e) {
-    e.hook("mount", e => {
-        e.ariaMultiLine = null;
-    }), e.hook("apply", (e, t) => {
-        "insert_text" === e.type ? e = {
-            ...e,
-            text: e.text.replaceAll("\n", "")
-        } : "insert_node" === e.type && (e = {
-            ...e,
-            fragment: [ A(...e.fragment) ]
-        }), t(e);
+function at(t) {
+    t.hook("mount", t => {
+        t.ariaMultiLine = null;
+    }), t.hook("apply", (t, e) => {
+        "insert_text" === t.type ? t = {
+            ...t,
+            text: t.text.replaceAll("\n", "")
+        } : "insert_node" === t.type && (t = {
+            ...t,
+            fragment: [ A(...t.fragment) ]
+        }), e(t);
     });
 }
 
-exports.Delete = function(t, n = e(t.selection)) {
-    t.apply({
+exports.Delete = function(e, n = t(e.selection)) {
+    e.apply({
         type: "delete",
         range: n
     });
-}, exports.Format = function(t, n, r, o = e(t.selection)) {
-    t.apply({
+}, exports.Format = function(e, n, r, o = t(e.selection)) {
+    e.apply({
         type: "format",
         range: o,
         key: n,
         value: r
     });
-}, exports.InsertNode = function(e, t, n = e.selection[0]) {
-    e.apply({
+}, exports.InsertNode = function(t, e, n = t.selection[0]) {
+    t.apply({
         type: "insert_node",
         at: n,
         fragment: [ {
-            children: [ t ]
+            children: [ e ]
         } ]
     });
-}, exports.InsertText = function(e, t, n = e.selection[0]) {
-    e.apply({
+}, exports.InsertText = function(t, e, n = t.selection[0]) {
+    t.apply({
         type: "insert_text",
         at: n,
-        text: t
+        text: e
     });
-}, exports.Redo = function(e) {
-    e.get(te).redo();
-}, exports.Redoable = function(e) {
-    return e.get(te).redoable();
-}, exports.ReplaceDoc = U, exports.ReplaceText = function(t, n) {
-    const r = e(t.selection);
-    t.apply({
+}, exports.Redo = function(t) {
+    t.get(et).redo();
+}, exports.Redoable = function(t) {
+    return t.get(et).redoable();
+}, exports.ReplaceDoc = U, exports.ReplaceText = function(e, n) {
+    const r = t(e.selection);
+    e.apply({
         type: "delete",
         range: r
     }).apply({
@@ -797,93 +807,93 @@ exports.Delete = function(t, n = e(t.selection)) {
         at: r[0],
         text: n
     });
-}, exports.SetBlockAttr = function(e, t, n, r = e.selection[0]) {
-    const {t: o} = R(e.doc, r);
-    e.apply({
+}, exports.SetBlockAttr = function(t, e, n, r = t.selection[0]) {
+    const {t: o} = R(t.doc, r);
+    t.apply({
         type: "set_node_attr",
         path: o,
-        key: t,
+        key: e,
         value: n
     });
-}, exports.ToggleBlockAttr = function(e, t, n, r, o = e.selection[0]) {
-    const {i: s, t: c} = R(e.doc, o);
-    e.apply({
+}, exports.ToggleBlockAttr = function(t, e, n, r, o = t.selection[0]) {
+    const {i: s, t: c} = R(t.doc, o);
+    t.apply({
         type: "set_node_attr",
         path: c,
-        key: t,
-        value: s[t] === n ? r : n
+        key: e,
+        value: s[e] === n ? r : n
     });
-}, exports.ToggleFormat = function(n, r, o = e(n.selection)) {
+}, exports.ToggleFormat = function(n, r, o = t(n.selection)) {
     let s;
-    if (t(o)) {
-        const e = ((e, t) => {
-            const n = R(e, t);
+    if (e(o)) {
+        const t = ((t, e) => {
+            const n = R(t, e);
             return O(n.i, n.o);
         })(n.doc, o[0]);
-        if (!e) return;
-        s = [ e.i ];
-    } else s = P(n.doc, ...o).flatMap(e => e.children);
-    const c = s.filter(x);
+        if (!t) return;
+        s = [ t.i ];
+    } else s = P(n.doc, ...o).flatMap(t => t.children);
+    const c = s.filter(y);
     c.length && n.apply({
         type: "format",
         range: o,
         key: r,
-        value: !!c.some(e => !e[r])
+        value: !!c.some(t => !t[r])
     });
-}, exports.Undo = function(e) {
-    e.get(te).undo();
-}, exports.Undoable = function(e) {
-    return e.get(te).undoable();
-}, exports.createEditor = se, exports.createPlainEditor = ({text: e, singleline: t, onChange: n, ...r}) => {
-    const o = se({
+}, exports.Undo = function(t) {
+    t.get(et).undo();
+}, exports.Undoable = function(t) {
+    return t.get(et).undoable();
+}, exports.createEditor = st, exports.createPlainEditor = ({text: t, singleline: e, onChange: n, ...r}) => {
+    const o = st({
         ...r,
         doc: {
-            children: l(e)
+            children: l(t)
         }
-    }).exec(ce);
-    return t && o.exec(ae), o.on("change", () => {
+    }).exec(ct);
+    return e && o.exec(at), o.on("change", () => {
         n(a(o.doc));
     }), o;
-}, exports.debugPlugin = function(e) {
-    e.hook("apply", e => {
-        console.log("apply", e);
-    }), e.on("change", () => {
-        console.log("change", e.doc);
-    }), e.on("selectionchange", () => {
-        console.log("selectionchange", e.selection);
+}, exports.debugPlugin = function(t) {
+    t.hook("apply", t => {
+        console.log("apply", t);
+    }), t.on("change", () => {
+        console.log("change", t.doc);
+    }), t.on("selectionchange", () => {
+        console.log("selectionchange", t.selection);
     });
-}, exports.fileTransferPlugin = (e, t) => {
-    e.hook("paste", e => {
-        for (const n of e.items) if ("file" === n.kind) {
-            const e = t[n.type];
-            if (e) {
-                const t = n.getAsFile();
-                if (t) return [ {
-                    children: [ e(t) ]
+}, exports.fileTransferPlugin = (t, e) => {
+    t.hook("paste", t => {
+        for (const n of t.items) if ("file" === n.kind) {
+            const t = e[n.type];
+            if (t) {
+                const e = n.getAsFile();
+                if (e) return [ {
+                    children: [ t(e) ]
                 } ];
             }
         }
         return null;
     });
-}, exports.htmlTransferPlugin = function(e, t) {
-    e.hook("mount", (n, r) => {
-        const o = e.hook("copy", e => {
-            const t = document.createElement("div");
-            t.appendChild(X(j(n), n).cloneContents()), e.setData("text/html", t.innerHTML);
-        }), s = e.hook("paste", e => {
-            const n = e.getData("text/html");
-            return n ? ((e, t, n) => {
-                const r = n.text, o = e => {
-                    const t = n[e.tagName.toLowerCase()];
-                    if (t) {
-                        const n = t(e);
+}, exports.htmlTransferPlugin = function(t, e) {
+    t.hook("mount", (n, r) => {
+        const o = t.hook("copy", t => {
+            const e = document.createElement("div");
+            e.appendChild(X(j(n), n).cloneContents()), t.setData("text/html", e.innerHTML);
+        }), s = t.hook("paste", t => {
+            const n = t.getData("text/html");
+            return n ? ((t, e, n) => {
+                const r = n.text, o = t => {
+                    const e = n[t.tagName.toLowerCase()];
+                    if (e) {
+                        const n = e(t);
                         if (n) return n;
                     }
                 };
-                let s = (new DOMParser).parseFromString(e, "text/html").body, c = !1;
+                let s = (new DOMParser).parseFromString(t, "text/html").body, c = !1;
                 for (const i of [ ...s.childNodes ]) V(i) ? "StartFragment" === i.data ? (c = !0, 
                 s = new DocumentFragment) : "EndFragment" === i.data && (c = !1) : c && s.appendChild(i);
-                return t(({u: e, h: t}) => {
+                return e(({u: t, h: e}) => {
                     let n, s = null, c = "", i = !1;
                     const a = [], l = () => {
                         c && (s || (s = []), s.push(r(c)), c = "");
@@ -892,37 +902,37 @@ exports.Delete = function(t, n = e(t.selection)) {
                             children: s
                         }), s = null, i = !1;
                     };
-                    for (;n = e(); ) if (4 === n) f(); else if (i = !0, 1 === n) c += t().data; else if (2 === n) {
+                    for (;n = t(); ) if (4 === n) f(); else if (i = !0, 1 === n) c += e().data; else if (2 === n) {
                         l();
-                        const e = o(t());
-                        e && s.push(e);
+                        const t = o(e());
+                        t && s.push(t);
                     } else 3 === n && f();
                     return f(), a.length || a.push({
                         children: []
                     }), a;
                 }, s);
-            })(n, r, t.serializers) : null;
+            })(n, r, e.serializers) : null;
         });
         return () => {
             o(), s();
         };
     });
-}, exports.internalTranferPlugin = function(t) {
-    t.hook("copy", n => {
-        n.setData(ie, JSON.stringify(P(t.doc, ...e(t.selection))));
-    }), t.hook("paste", e => {
+}, exports.internalTranferPlugin = function(e) {
+    e.hook("copy", n => {
+        n.setData(it, JSON.stringify(P(e.doc, ...t(e.selection))));
+    }), e.hook("paste", t => {
         try {
-            return JSON.parse(e.getData(ie));
-        } catch (t) {
+            return JSON.parse(t.getData(it));
+        } catch (e) {
             return null;
         }
     });
-}, exports.keymapPlugin = function(e, t) {
-    r(t).forEach(n => {
-        const r = t[n];
-        e.hook("keyboard", ee(n, () => {
+}, exports.keymapPlugin = function(t, e) {
+    r(e).forEach(n => {
+        const r = e[n];
+        t.hook("keyboard", tt(n, () => {
             r();
         }));
     });
-}, exports.plainTransferPlugin = ce, exports.singlelinePlugin = ae;
+}, exports.plainTransferPlugin = ct, exports.singlelinePlugin = at;
 //# sourceMappingURL=index.cjs.map
\ No newline at end of file
diff --git a/lib/index.js b/lib/index.js
index c55f72410d11dd4b27ce4e471dd0d6cb70cac2a8..9417645df05a60e59701a75a2a97ab481da06e1e 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -435,7 +435,7 @@ var ht = [], yt = () => {}, _t = t => {
         }));
     }, C = t => {
         if (!r) if (Array.isArray(t)) for (const e of t) U(e); else U(t);
-        return V;
+        return W;
     }, U = t => {
         const e = n, r = S("apply"), c = r.length;
         let s = 0;
@@ -514,7 +514,7 @@ var ht = [], yt = () => {}, _t = t => {
                         }
                         return [ t, e ];
                     })(n, x, t);
-                    (({type: t}) => t === d || t === p || t === h)(t) && !m(e) || (n = e, F(r));
+                    (({type: t}) => t === d || t === p || t === h)(t) && !m(e) || (n = e, V(r));
                 } catch (e) {
                     k("rollback operation: " + e);
                 }
@@ -523,10 +523,12 @@ var ht = [], yt = () => {}, _t = t => {
             e && (t = e), s++, i();
         };
         i(), o(e, n) || D("change");
-    }, F = t => {
+    };
+    let F;
+    const V = t => {
         !((t, [e, n]) => N(t, e) && N(t, n))(n, t) || x[0] === t[0] && x[1] === t[1] || (x = t, 
         D("selectionchange"));
-    }, V = {
+    }, W = {
         get doc() {
             return n;
         },
@@ -534,7 +536,7 @@ var ht = [], yt = () => {}, _t = t => {
             return x;
         },
         set selection(t) {
-            F(t);
+            V(t), F?.();
         },
         get readonly() {
             return r;
@@ -557,18 +559,18 @@ var ht = [], yt = () => {}, _t = t => {
             };
         },
         exec: (t, ...e) => {
-            const n = t(V, ...e);
-            return void 0 === n ? V : n;
+            const n = t(W, ...e);
+            return void 0 === n ? W : n;
         },
         get: t => (M.has(t) || g("No value found for key"), M.get(t)),
-        set: (t, e) => (M.set(t, e), V),
+        set: (t, e) => (M.set(t, e), W),
         input: o => {
             if (!window.InputEvent || !s(InputEvent.prototype.getTargetRanges)) return k("beforeinput event is not supported."), 
             yt;
             const {contentEditable: i, role: a, ariaMultiLine: l, ariaReadOnly: f} = o, u = o.style.whiteSpace;
             o.role = "textbox", o.style.whiteSpace = "pre-wrap", o.ariaMultiLine = "true";
-            let d = !1, p = null, h = !1, y = !1, g = !1;
-            const m = tt(o), v = ((t, e) => {
+            let d = !1, p = null, h = !1, y = !1, g = !1, m = !1;
+            const v = tt(o), M = ((t, e) => {
                 let n = null, r = null, o = null;
                 const c = () => {
                     if (null != o) return o;
@@ -635,23 +637,23 @@ var ht = [], yt = () => {}, _t = t => {
                     }
                 };
                 return l;
-            })(m, _), M = () => {
+            })(v, _), w = () => {
                 o.contentEditable = r ? "false" : "true", o.ariaReadOnly = r ? "true" : null;
             };
-            M();
-            const w = V.on("change", () => {
+            w();
+            const A = W.on("change", () => {
                 y || requestAnimationFrame(() => {
                     y || o.focus({
                         preventScroll: !0
                     });
                 });
-            }), A = V.on("readonly", M), O = t => {
+            }), O = W.on("readonly", w), R = t => {
                 for (const e of S("paste")) {
                     const n = e(t);
                     if (n) return n;
                 }
                 k("failed to serialize pasted data");
-            }, R = ((t, e) => {
+            }, D = ((t, e) => {
                 let n = !1, r = [];
                 const o = t => {
                     n && r.push(...t);
@@ -688,31 +690,39 @@ var ht = [], yt = () => {}, _t = t => {
                     }
                 };
             })(o, () => {
-                rt(m, o, v, T(n, x), x[0] - x[1]);
-            }), D = () => {
-                F(E(n, ((t, e) => {
+                rt(v, o, M, T(n, x), x[0] - x[1]);
+            }), B = () => {
+                V(E(n, ((t, e) => {
                     const n = et(t), r = nt(n, t);
                     if (!r) return [ [ [ 0 ], 0 ], [ [ 0 ], 0 ] ];
                     const o = st(t, e, r), c = $(n.anchorNode, n.focusNode);
                     return (0 === c ? n.anchorOffset > n.focusOffset : 2 & c) ? [ o[1], o[0] ] : o;
-                })(o, v)));
-            }, B = () => {
-                const t = R.A();
-                R.S(!1), t.length && (R.M(t), m.removeEventListener("selectionchange", G), rt(m, o, v, T(n, x), x[0] - x[1], !0), 
-                m.addEventListener("selectionchange", G)), p && (F(p[1]), C(p[0]), p = null), h = !1;
-            }, I = t => {
-                if (!h) for (const e of S("keyboard")) if (e(t)) return t.preventDefault(), void R.S(!1);
-            }, P = () => {
-                h || B();
-            }, N = t => {
+                })(o, M)));
+            };
+            F = () => {
+                !y || m || h || rt(v, o, M, T(n, x), x[0] - x[1]);
+            };
+            const I = () => {
+                m = !0;
+                const t = D.A();
+                D.S(!1), t.length && (D.M(t), v.removeEventListener("selectionchange", X), rt(v, o, M, T(n, x), x[0] - x[1], !0), 
+                v.addEventListener("selectionchange", X)), p && (V(p[1]), C(p[0]), p = null), h = !1, 
+                m = !1;
+            }, P = W.hook("apply", (t, e) => {
+                h && !m && I(), e();
+            }), N = t => {
+                if (!h) for (const e of S("keyboard")) if (e(t)) return t.preventDefault(), void D.S(!1);
+            }, H = () => {
+                h || I();
+            }, U = t => {
                 t.preventDefault();
                 const r = t.inputType;
                 if (r.startsWith("format")) return;
                 if ("historyUndo" === r || "historyRedo" === r) return;
-                h ? R.S(!0) : D();
+                h ? D.S(!0) : B();
                 const c = t.getTargetRanges()[0];
                 if (c) {
-                    const s = E(n, st(o, v, c));
+                    const s = E(n, st(o, M, c));
                     let i, a = "insertParagraph" === r || "insertLineBreak" === r ? "\n" : t.data;
                     if (null == a) {
                         const e = t.dataTransfer;
@@ -727,29 +737,29 @@ var ht = [], yt = () => {}, _t = t => {
                         text: a
                     });
                 }
-                h || B();
-            }, H = () => {
-                h || D(), h = !0;
-            }, U = () => {
-                B();
-            }, W = () => {
-                y = !0, D();
+                h || I();
             }, z = () => {
-                y = !1;
+                h || B(), h = !0;
             }, G = () => {
-                !y || h || g || D();
-            }, q = t => {
-                if (D(), !e(x)) for (const e of S("copy")) e(t);
-            }, J = t => {
-                t.preventDefault(), q(t.clipboardData);
-            }, X = e => {
-                e.preventDefault(), r || (q(e.clipboardData), C({
+                I();
+            }, q = () => {
+                y = !0, B();
+            }, J = () => {
+                y = !1;
+            }, X = () => {
+                !y || h || g || B();
+            }, Z = t => {
+                if (B(), !e(x)) for (const e of S("copy")) e(t);
+            }, K = t => {
+                t.preventDefault(), Z(t.clipboardData);
+            }, Q = e => {
+                e.preventDefault(), r || (Z(e.clipboardData), C({
                     type: "delete",
                     range: t(x)
                 }));
-            }, Z = e => {
+            }, ot = e => {
                 e.preventDefault();
-                const n = O(e.clipboardData);
+                const n = R(e.clipboardData);
                 if (n) {
                     const e = t(x), r = e[0], o = [ {
                         type: "delete",
@@ -765,7 +775,7 @@ var ht = [], yt = () => {}, _t = t => {
                         fragment: n
                     }), C(o);
                 }
-            }, K = e => {
+            }, it = e => {
                 e.preventDefault();
                 const r = e.dataTransfer, s = ((t, e, {clientX: n, clientY: r}) => {
                     const o = tt(t);
@@ -776,7 +786,7 @@ var ht = [], yt = () => {}, _t = t => {
                         const c = o.caretRangeFromPoint(n, r);
                         if (c) return ct(t, e, c.startContainer, c.startOffset);
                     }
-                })(o, v, e);
+                })(o, M, e);
                 if (r && s) {
                     let e;
                     const o = [];
@@ -784,7 +794,7 @@ var ht = [], yt = () => {}, _t = t => {
                         type: "delete",
                         range: t(x)
                     });
-                    const i = O(r);
+                    const i = R(r);
                     if (i) {
                         const t = b(n, s), r = L(t, o);
                         o.push(c(i) ? {
@@ -797,36 +807,36 @@ var ht = [], yt = () => {}, _t = t => {
                             fragment: i
                         }), e = [ r, L(t, o) ];
                     }
-                    C(o), e && F(e);
+                    C(o), e && V(e);
                 }
-            }, Q = t => {
-                g = !0, q(t.dataTransfer);
-            }, ot = () => {
+            }, at = t => {
+                g = !0, Z(t.dataTransfer);
+            }, lt = () => {
                 g = !1;
             };
-            m.addEventListener("selectionchange", G), o.addEventListener("keydown", I), o.addEventListener("input", P), 
-            o.addEventListener("beforeinput", N), o.addEventListener("compositionstart", H), 
-            o.addEventListener("compositionend", U), o.addEventListener("focus", W), o.addEventListener("blur", z), 
-            o.addEventListener("copy", J), o.addEventListener("cut", X), o.addEventListener("paste", Z), 
-            o.addEventListener("drop", K), o.addEventListener("dragstart", Q), o.addEventListener("dragend", ot);
-            const it = [];
+            v.addEventListener("selectionchange", X), o.addEventListener("keydown", N), o.addEventListener("input", H), 
+            o.addEventListener("beforeinput", U), o.addEventListener("compositionstart", z), 
+            o.addEventListener("compositionend", G), o.addEventListener("focus", q), o.addEventListener("blur", J), 
+            o.addEventListener("copy", K), o.addEventListener("cut", Q), o.addEventListener("paste", ot), 
+            o.addEventListener("drop", it), o.addEventListener("dragstart", at), o.addEventListener("dragend", lt);
+            const ft = [];
             return S("mount").forEach(t => {
-                const e = t(o, v);
-                e && it.push(e);
+                const e = t(o, M);
+                e && ft.push(e);
             }), () => {
-                d || (d = !0, w(), A(), o.contentEditable = i, o.role = a, o.ariaMultiLine = l, 
-                o.ariaReadOnly = f, o.style.whiteSpace = u, R.O(), m.removeEventListener("selectionchange", G), 
-                o.removeEventListener("keydown", I), o.removeEventListener("input", P), o.removeEventListener("beforeinput", N), 
-                o.removeEventListener("compositionstart", H), o.removeEventListener("compositionend", U), 
-                o.removeEventListener("focus", W), o.removeEventListener("blur", z), o.removeEventListener("copy", J), 
-                o.removeEventListener("cut", X), o.removeEventListener("paste", Z), o.removeEventListener("drop", K), 
-                o.removeEventListener("dragstart", Q), o.removeEventListener("dragend", ot), it.forEach(t => {
+                d || (d = !0, A(), O(), o.contentEditable = i, o.role = a, o.ariaMultiLine = l, 
+                o.ariaReadOnly = f, o.style.whiteSpace = u, D.O(), P(), F = void 0, v.removeEventListener("selectionchange", X), 
+                o.removeEventListener("keydown", N), o.removeEventListener("input", H), o.removeEventListener("beforeinput", U), 
+                o.removeEventListener("compositionstart", z), o.removeEventListener("compositionend", G), 
+                o.removeEventListener("focus", q), o.removeEventListener("blur", J), o.removeEventListener("copy", K), 
+                o.removeEventListener("cut", Q), o.removeEventListener("paste", ot), o.removeEventListener("drop", it), 
+                o.removeEventListener("dragstart", at), o.removeEventListener("dragend", lt), ft.forEach(t => {
                     t();
                 }));
             };
         }
     };
-    return V.exec(lt), V;
+    return W.exec(lt), W;
 }, gt = (t, e) => {
     t.hook("paste", t => {
         for (const n of t.items) if ("file" === n.kind) {
