diff --git a/dist/index.mjs b/dist/index.mjs
index 551ed20a066288d551e92e7e2b6ea2cd2a27cefb..8afe3dde1a51e33cb58380a1b1af63cd10e4bdc3 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -177,7 +177,7 @@ function createClient(clientOptions) {
       }
     }
     const contentLength = response.headers.get("Content-Length");
-    if (response.status === 204 || request.method === "HEAD" || contentLength === "0" && !response.headers.get("Transfer-Encoding")?.includes("chunked")) {
+    if (response.status === 204 || request.method === "HEAD") {
       return response.ok ? { data: void 0, response } : { error: void 0, response };
     }
     if (response.ok) {
