diff --git a/dist/index.mjs b/dist/index.mjs
index 0a380f1a9f13e5feae40ae08349d2f29632e2856..e7425dd4f08f8c9044d409825200c18a415756e1 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -170,7 +170,7 @@ function createClient(clientOptions) {
         }
       }
     }
-    if (response.status === 204 || request.method === "HEAD" || response.headers.get("Content-Length") === "0") {
+    if (response.status === 204 || request.method === "HEAD") {
       return response.ok ? { data: void 0, response } : { error: void 0, response };
     }
     if (response.ok) {
