diff --git a/dist/index.mjs b/dist/index.mjs
index 724ebef59d87aa57d240b233c73f6922ac609007..12ab5992b9461655555b238296dac275d9c32157 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -170,7 +170,7 @@ function createClient(clientOptions) {
         }
       }
     }
-    if (response.status === 204 || request.method === "HEAD" || response.headers.get("Content-Length") === "0") {
+    if (response.status === 204 || request.method === "HEAD") {
       return response.ok ? { data: void 0, response } : { error: void 0, response };
     }
     if (response.ok) {
