diff --git a/lib/index.cjs b/lib/index.cjs
index 74f14560f50b0e578d531932445b3e43d80e9da8..77a546a8199afc37b7bf56c1dc3cc2716383bc0b 100644
--- a/lib/index.cjs
+++ b/lib/index.cjs
@@ -235,6 +235,9 @@ var {min: a, max: d, abs: h, floor: g} = Math, v = (e, t, o) => a(o, d(t, e)), S
             t.removeEventListener("scroll", v), t.removeEventListener("wheel", S), t.removeEventListener("touchstart", m), 
             t.removeEventListener("touchend", w), $(s);
         },
+        Z: () => {
+            n() !== e.C() && v();
+        },
         Y: () => {
             const [t, o] = e.j();
             t && (r(t, o, a), a = !1, o && e.o() > e.t() && e.q(1, n()));
@@ -277,12 +280,12 @@ var {min: a, max: d, abs: h, floor: g} = Math, v = (e, t, o) => a(o, d(t, e)), S
             behavior: "smooth"
         }) : o[s] = e;
     }), u = t ? "width" : "height", a = /* @__PURE__ */ new WeakMap, d = E(t => {
-        const n = [];
-        for (const {target: r, contentRect: i} of t) if (r.offsetParent) if (r === o) e.q(4, i[u]); else {
+        const h = [];
+        for (const {target: r, contentRect: i} of t) if (r.offsetParent) if (r === o) n && n.Z(), e.q(4, i[u]); else {
             const e = a.get(r);
-            null != e && n.push([ e, i[u] ]);
+            null != e && h.push([ e, i[u] ]);
         }
-        n.length && e.q(3, n);
+        h.length && e.q(3, h);
     });
     return {
         A(c, u = c.parentElement) {
@@ -381,10 +384,10 @@ var {min: a, max: d, abs: h, floor: g} = Math, v = (e, t, o) => a(o, d(t, e)), S
             top: e
         };
         t && (n.behavior = "smooth"), o.scrollTo(n);
-    }), a = /* @__PURE__ */ new WeakMap, h = /* @__PURE__ */ new Set, g = /* @__PURE__ */ new Set, v = /* @__PURE__ */ new Map, S = (e, t) => `${e}-${t}`, $ = E(n => {
+    }), a = /* @__PURE__ */ new WeakMap, h = /* @__PURE__ */ new Set, g = /* @__PURE__ */ new Set, v = /* @__PURE__ */ new Map, S = (e, t) => `${e}-${t}`, $ = E(x => {
         const r = /* @__PURE__ */ new Set, i = /* @__PURE__ */ new Set;
-        for (const {target: s, contentRect: {width: l, height: c}} of n) if (s.offsetParent) if (s === o) e.q(4, c), 
-        t.q(4, l); else {
+        for (const {target: s, contentRect: {width: l, height: c}} of x) if (s.offsetParent) if (s === o) n && n.Z(), 
+        r && r.Z(), e.q(4, c), t.q(4, l); else {
             const e = a.get(s);
             if (e) {
                 const [t, o] = e, n = S(t, o), s = v.get(n);
diff --git a/lib/index.js b/lib/index.js
index 48b62dedcb4c0951eb1563c5e86974c43fc9f71e..d5b776cb5ffbbfa543666e2155aca496767f3fae 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -217,6 +217,9 @@ var {min: u, max: a, abs: d, floor: h} = Math, g = (e, t, o) => u(o, a(t, e)), v
             t.removeEventListener("scroll", v), t.removeEventListener("wheel", m), t.removeEventListener("touchstart", p), 
             t.removeEventListener("touchend", w), $(s);
         },
+        S: () => {
+            n() !== e.T() && v();
+        },
         A: () => {
             const [t, o] = e.L();
             t && (r(t, o, a), a = !1, o && e.o() > e.t() && e.W(1, n()));
@@ -259,12 +262,12 @@ var {min: u, max: a, abs: d, floor: h} = Math, g = (e, t, o) => u(o, a(t, e)), v
             behavior: "smooth"
         }) : o[s] = e;
     }), u = t ? "width" : "height", a = /* @__PURE__ */ new WeakMap, d = O(t => {
-        const n = [];
-        for (const {target: r, contentRect: i} of t) if (r.offsetParent) if (r === o) e.W(4, i[u]); else {
+        const h = [];
+        for (const {target: r, contentRect: i} of t) if (r.offsetParent) if (r === o) n && n.S(), e.W(4, i[u]); else {
             const e = a.get(r);
-            null != e && n.push([ e, i[u] ]);
+            null != e && h.push([ e, i[u] ]);
         }
-        n.length && e.W(3, n);
+        h.length && e.W(3, h);
     });
     return {
         D(c, u = c.parentElement) {
@@ -363,10 +366,10 @@ var {min: u, max: a, abs: d, floor: h} = Math, g = (e, t, o) => u(o, a(t, e)), v
             top: e
         };
         t && (n.behavior = "smooth"), o.scrollTo(n);
-    }), d = /* @__PURE__ */ new WeakMap, h = /* @__PURE__ */ new Set, g = /* @__PURE__ */ new Set, v = /* @__PURE__ */ new Map, $ = (e, t) => `${e}-${t}`, m = O(n => {
+    }), d = /* @__PURE__ */ new WeakMap, h = /* @__PURE__ */ new Set, g = /* @__PURE__ */ new Set, v = /* @__PURE__ */ new Map, $ = (e, t) => `${e}-${t}`, m = O(x => {
         const r = /* @__PURE__ */ new Set, i = /* @__PURE__ */ new Set;
-        for (const {target: s, contentRect: {width: l, height: c}} of n) if (s.offsetParent) if (s === o) e.W(4, c), 
-        t.W(4, l); else {
+        for (const {target: s, contentRect: {width: l, height: c}} of x) if (s.offsetParent) if (s === o) n && n.S(), 
+        r && r.S(), e.W(4, c), t.W(4, l); else {
             const e = d.get(s);
             if (e) {
                 const [t, o] = e, n = $(t, o), s = v.get(n);
