diff --git a/dist/types/components/searchbar/searchbar.d.ts b/dist/types/components/searchbar/searchbar.d.ts
index 88d3eb8cd927231f2b418608c3bc8cd7963f8d3d..73a8b4cbdede4249454e60d43a007d0c4dd75472 100644
--- a/dist/types/components/searchbar/searchbar.d.ts
+++ b/dist/types/components/searchbar/searchbar.d.ts
@@ -55,7 +55,7 @@ export declare class Searchbar implements ComponentInterface {
      * Indicates whether and how the text value should be automatically capitalized as it is entered/edited by the user.
      * Available options: `"off"`, `"none"`, `"on"`, `"sentences"`, `"words"`, `"characters"`.
      */
-    autocapitalize: string;
+    autocapitalize?: string;
     /**
      * Set the input's autocomplete property.
      */
diff --git a/dist/types/components.d.ts b/dist/types/components.d.ts
index 50a467eea4d61a0a37776335505954ed886de09f..577fc3f8fae5e7e3f6539975e25eee18a0eb5643 100644
--- a/dist/types/components.d.ts
+++ b/dist/types/components.d.ts
@@ -7114,7 +7114,7 @@ declare namespace LocalJSX {
         /**
           * Indicates whether and how the text value should be automatically capitalized as it is entered/edited by the user. Available options: `"off"`, `"none"`, `"on"`, `"sentences"`, `"words"`, `"characters"`.
          */
-        "autocapitalize": string;
+        "autocapitalize"?: string;
         /**
           * Set the input's autocomplete property.
          */
